pkg/controller/volume/persistentvolume/controller_base.go

```diff
@@ -47,13 +48,13 @@ import (
 func NewPersistentVolumeController(
 	kubeClient clientset.Interface,
 	syncPeriod time.Duration,
-	provisioner vol.ProvisionableVolumePlugin,
-	recyclers []vol.VolumePlugin,
+	volumePlugins []vol.VolumePlugin,
 	cloud cloudprovider.Interface,
 	clusterName string,
-	volumeSource, claimSource cache.ListerWatcher,
+	volumeSource, claimSource, classSource cache.ListerWatcher,
 	eventRecorder record.EventRecorder,
 	enableDynamicProvisioning bool,
+	defaultStorageClass string,
```

Comment:
AFAICT this is always "" - can you explain why we need it?  I'll write more on defaults in a bit.